### PR TITLE
chore: Browser E2E tests for compass-web sandbox COMPASS-7598 COMPASS-7599

### PR DESCRIPTION
**Pull request by lerouxb**
Extracted from https://github.com/mongodb-js/compass/pull/5355

This only runs the time before first query tests in compass-web / the browser. Future tickets will un-skip more tests.

I was going to add it to CI in a separate PR, but then figured it is better to prove that it works and the change for that is quite small.

There are two new npm run scripts in compass-e2e-tests:

```
    "test-web": "env DEBUG=hadron*,mongo*,compass* npm run test -- --test-compass-web",
    "test-web-noserver": "env DEBUG=hadron*,mongo*,compass* npm run test -- --test-compass-web --disable-start-stop --bail",
```

`test-web` just mirrors `test` and it will run some e2e tests against the compass-web sandbox. It starts the database using mongodb-runner and it starts compass-web itself. Then closes those at the end.
`test-web-noserverver` won't start the database or compass-web and won't stop them. It also bails on the first test failure. This is useful during e2e test development for maximum speed where you can start these things once and keep them running.

These commands also support the new env var BROWSER_NAME which defaults to "chrome". The other value that's currently supported is "firefox". This picks the browser to use in testing.

And then this PR also adds two new tasks (test-web-sandbox-chrome and test-web-sandbox-firefox) to the ubuntu build variant to run these tests in chrome and firefox. We're only doing the ubuntu variant for now because we don't have any platform differences in compass-web yet.

### CODE REVIEW

File: packages/compass-e2e-tests/tests/atlas-login.test.ts [hunk 1]
```diff
@@ -121,6 +125,10 @@ describe('Atlas Login', function () {
   });
 
   after(async function () {
+    if (TEST_COMPASS_WEB) {
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/atlas-login.test.ts:128]: Unfortunately mocha still calls after() if you this.skip() in before() and that's by design: https://github.com/mochajs/mocha/issues/3120

File: packages/compass-e2e-tests/helpers/compass.ts [hunk 2]
```diff
@@ -631,14 +679,26 @@ async function startCompass(
             .join(', ')}`
     );
 
-    for (const p of filteredProcesses) {
-      tryKillProcess(p.pid, p.name);
```
Comment (lerouxb) [inline packages/compass-e2e-tests/helpers/compass.ts:635]: oops i think I overrode this by accident. Will fix.

File: packages/compass-e2e-tests/index.ts [hunk 3]
```diff
@@ -244,6 +295,9 @@ async function run() {
 
     cleanup();
   }
+
+  // TODO: why does this somehow hang? How do you debug that?
+  process.exit(process.exitCode ?? 0);
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/index.ts:300]: [`why-is-node-running`](https://www.npmjs.com/package/why-is-node-running) is a good starting point to investigate stuff like this

Comment (lerouxb) [reply packages/compass-e2e-tests/index.ts]: Oof it had an error while trying to copy logs when stopped that was swallowed and I never noticed. So it never actually stopped compass properly.

File: packages/compass-e2e-tests/helpers/commands/wait-for-connection-result.ts [hunk 4]
```diff
@@ -8,13 +9,18 @@ export async function waitForConnectionResult(
...
   } else if (connectionStatus === 'success') {
     // First meaningful thing on the screen after being connected, good enough
     // indicator that we are connected to the server
-    selector = Selectors.MyQueriesList;
+    // TODO: the web selector might actually work for both cases
+    selector = TEST_COMPASS_WEB
+      ? '[data-testid="workspace-tab-button"][title=Databases]'
+      : Selectors.MyQueriesList;
   } else {
+    // TODO: this doesn't support compass-web yet, but also isn't encountered yet
```
Comment (addaleax) [inline packages/compass-e2e-tests/helpers/commands/wait-for-connection-result.ts:23]: Which TODOs are meant to be addressed in this PR and which are going to be part of follow-up work?

File: packages/compass-e2e-tests/index.ts [hunk 5]
```diff
@@ -18,9 +22,13 @@ import {
 } from './helpers/compass';
 import ResultLogger from './helpers/result-logger';
 
+// https://github.com/nodejs/node/issues/40537
+dns.setDefaultResultOrder('ipv4first');
```
Comment (addaleax) [inline packages/compass-e2e-tests/index.ts:26]: We probably shouldn't be introducing new usage of this hack and instead address the actual issue? We only accepted that we'd need to add it in a few places in order to get things working quickly, not as a real fix

Comment (lerouxb) [reply packages/compass-e2e-tests/index.ts:26]: How do you fix the real issue?

Comment (addaleax) [reply packages/compass-e2e-tests/index.ts:27]: > How do you fix the real issue? 

Depends on what we're trying to do/where it's failing – if we want to connect to a MongoDB server that's listening on `127.0.0.1` only, we should connect to it using `127.0.0.1`, not `localhost`.

> The ideal there (assuming node's own dns module isn't going to do it) would be for the driver to switch out to a dns module that does, right?

No, for connections made using the Node.js driver, it just needs to implement the spec and connect to all addresses returned from `getaddrinfo`; https://jira.mongodb.org/browse/NODE-5754 tracks that (the title is a bit misleading in the sense that this isn't about IPv4 vs IPv6 specifically; the "I want the driver to attempt to use all A and AAAA records for each hostname" bit is the relevant one).

File: packages/compass/src/main/handle-unhandled-rejection.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import { app, dialog, clipboard } from 'electron';
+import cleanStack from 'clean-stack';
+import ensureError from 'ensure-error';
+import COMPASS_ICON from './icon';
+
+async function handleUnhandledRejection(err: Error): Promise<void> {
```
Comment (lerouxb) [inline packages/compass/src/main/handle-unhandled-rejection.ts:6]: Right now this is just a copy of the uncaught exception handler with literally only the console.error line changed.

**Pull request merged**